wdl-lint/src/rules/expression_spacing.rs

```diff
@@ -0,0 +1,726 @@
+    Diagnostic::note("multi-line if...then...else must have a preceding parenthesis and newline")
+        .with_rule(ID)
+        .with_highlight(span)
+        .with_fix("add a open parenthesis and newline prior to this if")
+}
+
+/// Reports missing newline prior to then in multiline if...then...else
+/// constructs
+fn multiline_then_space(span: Span) -> Diagnostic {
+    Diagnostic::note("multi-line if...then...else must have a preceding space")
+        .with_rule(ID)
+        .with_highlight(span)
+        .with_fix("add a newline before the then keyword")
+}
+
+/// Reports missing newline prior to else in multiline if...then...else
+/// constructs
+fn multiline_else_space(span: Span) -> Diagnostic {
+    Diagnostic::note("multi-line if...then...else must have a preceding space")
+        .with_rule(ID)
+        .with_highlight(span)
+        .with_fix("add a newline before the else keyword")
+}
+
+/// Reports missing close paren for multiline if...then...else constructs
+fn multiline_if_close_paren(span: Span) -> Diagnostic {
+    Diagnostic::note("multi-line if...then...else must have a following newline and parenthesis")
+        .with_rule(ID)
+        .with_highlight(span)
+        .with_fix("add a newline and close parenthesis after to this else clause")
+}
+
+/// Reports missing newline following open brace/bracket/paren for multiline
+/// array/map/object literals
+fn multiline_literal_open_newline(span: Span) -> Diagnostic {
+    Diagnostic::note(
+        "multi-line array/map/object literals must have a newline following the opening token",
+    )
+    .with_rule(ID)
+    .with_highlight(span)
+    .with_fix("add a newline after this token")
+}
+
+/// Reports missing newline before close brace/bracket/paren for multiline
+/// array/map/object literals
+fn multiline_literal_close_newline(span: Span) -> Diagnostic {
+    Diagnostic::note(
+        "multi-line array/map/object literals must have a newline preceding the closing token",
+    )
+    .with_rule(ID)
+    .with_highlight(span)
+    .with_fix("add a newline before this token")
+}
+
+/// Detects improperly spaced expressions.
+#[derive(Default, Debug, Clone, Copy)]
+pub struct ExpressionSpacingRule;
+
+impl Rule for ExpressionSpacingRule {
+    fn id(&self) -> &'static str {
+        ID
+    }
+
+    fn description(&self) -> &'static str {
+        "Ensures that WDL expressions are properly spaced."
+    }
+
+    fn explanation(&self) -> &'static str {
+        "Proper spacing is important for readability and consistency. This rule ensures that \
+         expressions are spaced properly. The following tokens should be surrounded by whitespace \
+         when used as an infix: `=`, `==`, `!=`, `&&`, `||`, `<`, `<=`, `>`, `>=`, `+`, `-`, `*`, \
+         `/`, and `%`. The following tokens should not be followed by whitespace when used as a \
+         prefix: `+`, `-`, and `!`. Opening brackets (`(`, `[`) should not be followed by a space, \
```

Comment:
I believe you pointed out over slack that in WDL `+` can't be used as a prefix operator? It should be removed from this clause here.